kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/interceptors/TestCoroutineInterceptor.kt

```diff
@@ -26,11 +28,21 @@ class TestCoroutineInterceptor : TestExecutionInterceptor {
    ): TestResult {
       var result: TestResult = TestResult.Ignored
       logger.log { Pair(testCase.name.testName, "Switching context to coroutines runTest") }
-      runTest {
+
+      val testBody: suspend CoroutinesTestScope.() -> Unit = {
          withContext(TestScopeElement(this)) {
             result = test(testCase, scope.withCoroutineContext(coroutineContext))
          }
       }
+
+      // Handle invocation timeout here, not in InvocationTimeoutInterceptor
+      // to avoid the influence of default timeout set inside runTest
+      val timeout = testCase.config.invocationTimeout
+      if (timeout != null && testCase.type != TestType.Container) {
+         runTest(timeout = timeout, testBody = testBody)
+      } else {
+         runTest(testBody = testBody)
+      }
```

Comment:
I know that this is unrelated to this PR, but this `runTest` invocation occurs in common code, and its test result is discarded instead of being returned immediately to the encompassing (JS) test framework. [This is described to cause losing test errors.](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-test/kotlinx.coroutines.test/-test-result/)

Reply:
As I know, currently it is impossible to use `TestCoroutinesInterceptor` with JS. It is added only if this condition is true:
```kotlin
val useCoroutineTestScope = platform != Platform.JS && testCase.config.coroutineTestScope
```
So it doesn't affect JS tests yet.

Reply:
That makes sense. Thanks for pointing it out. Maybe you could grab the opportunity and [add a comment why `coroutineTestScope` is excluded on JS](https://github.com/kotest/kotest/blob/85f6370c608c955030260dd14d393730b3834829/kotest-framework/kotest-framework-engine/src/commonMain/kotlin/io/kotest/engine/test/TestCaseExecutor.kt#L83).

Reply:
Should I also create an issue to track this change?

Reply:
That would be super nice!

Reply:
Here it is: https://github.com/kotest/kotest/issues/4077

Reply:
Added this comment in separate PR: https://github.com/kotest/kotest/pull/4078
To make this change visible in changelogs.